packages/pangea-sdk/pangea/services/intel.py

```diff
@@ -1073,3 +1074,26 @@ def password_breached(
         response = self.request.post("password/breached", data=input.dict(exclude_none=True))
         response.result = UserPasswordBreachedResult(**response.raw_result)
         return response
+
+    class PasswordStatus(enum.Enum):
+        BREACHED = 0
+        UNBREACHED = 1
+        INCONCLUSIVE = 2
+
+    @staticmethod
+    def is_password_breached(response: PangeaResponse[UserBreachedResult], hash: str) -> PasswordStatus:
+        if response.result.raw_data is None:
+            raise PangeaException("Need raw data to check if hash is breached. Send request with raw=true")
+
+        hash_data = response.result.raw_data.pop(hash, None)
+        if hash_data is not None:
+            # If hash is present in raw data, it's because it was breached
+            return UserIntel.PasswordStatus.BREACHED
+        else:
+            # If it's not present, should check if I have all breached hash
+            # Server will return a maximum of 1000 hash, so if breached count is greater than that,
+            # I can't conclude is password is or is not breached
+            if len(response.result.raw_data.keys()) >= 1000:
+                return UserIntel.PasswordStatus.INCONCLUSIVE
```

Comment:
same here again, need to check each entry for match.

Reply:
If I'm not wrong, raw result has a dictionary where full hash is the key and value is an object which has "count" information. So, what I did is to check if my full hash is present on that dictionary, if it's not present, I check the number of keys to decide.